subiquity/models/tests/test_filesystem.py

```diff
@@ -1043,3 +1046,29 @@ def test_lv_remaninig(self):
         lv2 = model._one(type="lvm_partition", id='lv2')
         self.assertEqual(
             lv2.size, vg.available_for_partitions - dehumanize_size("50M"))
+
+    def test_render_does_not_include_unreferenced(self):
+        model = make_model(Bootloader.NONE)
+        disk1 = make_disk(model, preserve=True)
+        disk2 = make_disk(model, preserve=True)
+        disk1p1 = make_partition(model, disk1, preserve=True)
+        disk2p1 = make_partition(model, disk2, preserve=True)
+        fs = model.add_filesystem(disk1p1, 'ext4')
+        model.add_mount(fs, '/')
+        rendered_ids = {action['id'] for action in model._render_actions()}
+        self.assertTrue(disk1.id in rendered_ids)
+        self.assertTrue(disk1p1.id in rendered_ids)
+        self.assertTrue(disk2.id not in rendered_ids)
+        self.assertTrue(disk2p1.id not in rendered_ids)
+
```

Comment:
Does this mean that users can create partitions in the UI and if they don't format or mount them we'd not include that in the storage config?  I suspect users would not be happy about that.  Or are you just constructed probed partitions which were already present but not used?

Reply:
> Does this mean that users can create partitions in the UI and if they don't format or mount them we'd not include that in the storage config? I suspect users would not be happy about that.

No, those partitions would have preserve=False and so be included. I can add a test to be explicit about this.

> Or are you just constructed probed partitions which were already present but not used?

Right.